docs/configuration.md

```diff
@@ -0,0 +1,91 @@
+# Configuration
+
+### allowUnused
+array of strings or regular expressions to match against a module name
+to determine if it is allowed to be unused
+
+Please create an [issue](https://github.com/charlierudolph/dependency-lint/issues)
+anytime you need to use this
+
+default:
+```yml
+allowUnused: []
```

Comment:
I would say no need to provide the default values here. (A) to me "default" means what the system uses if this isn't provided here. So the default file should be empty. (B) The user sees the "default" values when she looks at the `dependency-lint.yml` file that is generated for her. 

What is more useful here in terms of documenting things would be some representative examples. With those, you can omit describing the type in the comments. This (A) makes the comments shorter and more readable, and (B) gives some practical examples how exactly it looks like if I want to provide values here.

```
### allowUnused

List of modules that should be ignored by dependency-lint.

Please create an [issue](https://github.com/charlierudolph/dependency-lint/issues)
anytime you need to use this

Examle:

allowUnused:
  - mocha
```